src/realm/object-store/sync/generic_network_transport.cpp

```diff
@@ -61,19 +61,29 @@ const char* httpmethod_to_string(HttpMethod method)
     return "UNKNOWN";
 }
 
-AppError::AppError(ErrorCodes::Error error_code, std::string message, std::string link,
-                   util::Optional<int> additional_error_code)
-    : RuntimeError(error_code,
-                   error_code == ErrorCodes::HTTPError ? http_message(message, *additional_error_code) : message)
+AppError::AppError(ErrorCodes::Error code, std::string message, std::string link,
+                   std::optional<int> additional_error_code)
+    : Exception(code, code == ErrorCodes::HTTPError ? http_message(message, *additional_error_code) : message)
     , additional_status_code(additional_error_code)
     , link_to_server_logs(link)
 {
-    REALM_ASSERT(ErrorCodes::error_categories(error_code).test(ErrorCategory::app_error));
+    // For these errors, the server_error string is empty
+    REALM_ASSERT(ErrorCodes::error_categories(code).test(ErrorCategory::app_error));
+}
+
+AppError::AppError(std::string server_err, std::string message, std::string link,
```

Comment:
you could make `server_err` optional and then have only one constructor

Reply:
Updated